### ISSUE TITLE
Suggest read only mode should bcast transaction

### ISSUE THREAD

**Issue by firesWu** [EOSIO/eos#7832]
When node is read only mode, it will drop any trx and don't bcast transaction. If my node only connect read-only node (Unfortunately, my p2p list is all read-only node), so that I can't push any transaction arrive mainnet, because it will drop when it arrive read-only node. We can't know it is a read-only node from peer info, so I can't know why my trx can't arrive mainnet by my node. So I suggest read-only mode should bcast trx.

```
void net_plugin_impl::handle_message(const connection_ptr& c, const packed_transaction_ptr& trx) {
   //...

   // if mode is READ_ONLY , return
   if( cc.get_read_mode() == eosio::db_read_mode::READ_ONLY ) {
      fc_dlog(logger, "got a txn in read-only mode - dropping");
      return;
   }
   
   //...
}
```

**Comment by heifner** [comment]
We don't want read-only to broadcast a transaction because it has not validated it. To validate it would require state modifications making it non-read-only. 

The chain api sends back a "Not allowed, node in read-only mode" error when pushing a transaction to a read-only api node.

Maybe `get_info` could indicate the mode of the node.

**Comment by firesWu** [comment]
> indicate

I realized that it will lead to more junk trx for read-only node.  
 Maybe indicate the mode info in `v1/net/connections` .  peer maybe hasn't open http point.

**Comment by heifner** [comment]
I don't think we would object to a PR with that addition to `v1/net/connections`
